project_leverage/controllers/main.py

```diff
@@ -0,0 +1,101 @@
+# Copyright (C) 2022 Open Source Integrators
+# Copyright (C) 2020 Serpent Consulting Services
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo.http import request
+
+from odoo.addons.sale_timesheet.controllers.portal import SaleTimesheetCustomerPortal
+
+DEFAULT_MONTH_RANGE = 3
+
+
+class SaleTimesheetController(SaleTimesheetCustomerPortal):
+    def _plan_prepare_values(self, projects):
+        employee_obj = request.env["hr.employee"]
+        values = super(SaleTimesheetController, self)._plan_prepare_values(projects)
+        _select = self.select_line()
+        _from = self.line_from()
+        _where, param = self.line_where(projects)
+        _group = self.line_group()
+        query = _select + _from + _where + _group
+        request.env.cr.execute(query, param)
+        raw_data = request.env.cr.dictfetchall()
+        project_data = {}
+        month_list = []
+        month_name_list = []
+        for rec in raw_data:
+            month_name_list.append(rec.get("month_year"))
+            if (
+                str(rec.get("date_mm")).strip() + str(int(rec.get("year")))
+                not in month_list
+            ):
+                month_list.append(
+                    str(rec.get("date_mm")).strip() + str(int(rec.get("year")))
```

Comment:
would be so nice to have this string computed only once. Why do we call str()? concatenating 2 strings will return a string. 

You will get a crash if date_mm is not in rec (because rec.get() will return None and the call to strip() will fail. Same if "year" is not in rec. If you are sure the key is present, then don't use get, use a normal key access, otherwise use a default value which will not crash strip() or int() respectively.